arch/arc/core/irq_manage.c

```diff
@@ -83,6 +83,16 @@ void _irq_priority_set(unsigned int irq, unsigned int prio, u32_t flags)
 
 	__ASSERT(prio < CONFIG_NUM_IRQ_PRIO_LEVELS,
 		 "invalid priority %d for irq %d", prio, irq);
+/* 0 -> CONFIG_NUM_IRQ_PRIO_LEVELS allocted to secure world
```

Comment:
is this split required? This enforces secure priorities are higher than normal priorities. And, it requires hardware configuration with multiple priority levels (guess most ARC configs would have that anyway).

Reply:
I think it's required, or the normal mode can attack the secure mode through interrupt. With this assumption, the whole system is more secure and simpler, e.g. the cases of bounce of interrupts between secure and normal.  It's just something like fast irq's priority is higher then regular irq

Reply:
Interrupt priorities are typically assigned based on real-time response requirements. That can be different from security concerns. Fast vs regular IRQ's is about response times, so different.